packages/pwa-kit-dev/src/configs/webpack/overrides-plugin.js

```diff
@@ -0,0 +1,141 @@
+        this.projectDir = options.projectDir
+        this._allSearchDirs = [this.projectDir + this.overridesDir, ...this.extends]
+        this.pkg = require(path.resolve(this.projectDir, 'package.json'))
+        this.extendsHashMap = new Map()
+
+        const OVERRIDES_EXTENSIONS = '.+(js|jsx|ts|tsx|svg|jpg|jpeg)'
+        const globPattern = `${this.pkg?.ccExtensibility?.overridesDir?.replace(
+            /\//,
+            ''
+        )}/**/*${OVERRIDES_EXTENSIONS}`
+        const overridesFsRead = glob.sync(globPattern)
+
+        const overrideReplace = this.pkg?.ccExtensibility?.overridesDir + '/'
+
+        overridesFsRead.forEach((item) => {
+            const end = item.substring(item.lastIndexOf('/index'))
+            const [l, ...rest] = item.split(/(index|\.)/)
+            this.extendsHashMap.set(
+                l.replace(/\/$/, '')?.replace(overrideReplace.replace(/\//, ''), ''),
+                [end, rest]
+            )
+        })
+    }
+
+    /**
+     *
+     * @param requestPath
+     * @param dirs
+     */
+    findFileMap(requestPath, dirs) {
+        var fileExt = path.extname(requestPath)
+        for (var dir of dirs) {
+            var base = path.join(dir, requestPath)
+            if (fileExt) {
+                const noExtPath = requestPath.replace(fileExt, '')
+                if (this.extendsHashMap.has(noExtPath)) {
+                    return base
+                }
+            } else {
+                if (this.extendsHashMap.has(requestPath)) {
+                    const end = this.extendsHashMap.get(requestPath)[1]
+
+                    if (end[0] === 'index') {
+                        base = path.join(base, this.extendsHashMap.get(requestPath)[1].join(''))
+                        return base
+                    } else {
+                        base = base?.replace(/$\//, '') + end.join('')
+                        return base
+                    }
+                }
+            }
+        }
+    }
+
+    toOverrideRelative(path) {
+        var override = this.findOverride(path)
+        return path.substring(override.length + 1)
+    }
+
+    findOverride(path) {
+        return this._allSearchDirs.find((override) => {
+            return path.indexOf(override) === 0
+        })
+    }
+
+    isFromExtends(request, path) {
+        return (
+            request &&
+            this.extends.includes(request?.split('/')?.[0]) &&
+            // this is very important, to avoid circular imports, check that the
+            // `issuer` (requesting context) isn't the overrides directory
+            !path.match(this.projectDir + this.overridesDir)
+        )
+    }
+
+    apply(resolver) {
+        resolver.getHook('resolve').tapAsync(
+            'FeatureResolverPlugin',
+            function (requestContext, resolveContext, callback) {
+                if (
+                    requestContext.request &&
+                    this.isFromExtends(requestContext.request, requestContext.path)
```

Comment:
the basic check logic is "is this file coming from extends" = "is the requesting context from within `retail-react-app`